python/baseline/services.py

```diff
@@ -273,21 +328,42 @@ def predict(self, tokens, **kwargs):
 
         """
         preproc = kwargs.get('preproc', 'client')
+        exporter_field_feature_map = kwargs.get('exporter_field_feature_map', {})  # if empty dict argument was passed
+        if not exporter_field_feature_map:
+            exporter_field_feature_map = {'tokens': 'text'}
         label_field = kwargs.get('label', 'label')
         tokens_seq, mxlen, mxwlen = self.batch_input(tokens)
         self.set_vectorizer_lens(mxlen, mxwlen)
+        # TODO: unlike classify, here we allow vectorizers even for preproc=server to get `word_lengths`.
+        # vectorizers should not be available when preproc=server.
         examples = self.vectorize(tokens_seq)
         if preproc == 'server':
-            examples['tokens'] = [" ".join([y['text'] for y in x]) for x in tokens_seq]
+            unfeaturized_examples = {}
+            for exporter_field in exporter_field_feature_map:
+                unfeaturized_examples[exporter_field] = [" ".join([y[exporter_field_feature_map[exporter_field]]
+                                                                   for y in x]) for x in tokens_seq]
+            unfeaturized_examples['word_lengths'] = examples['word_lengths']
```

Comment:
This seems like it should check for some key to use for lengths rather then just defaulting `word_lengths`, I could be wrong though

Reply:
yes we could i think